src/generator/polymorphics.ts

```diff
@@ -57,7 +57,12 @@ export async function generatePolymorphicHelpers(session: Session<CodeModel>): P
     text += `\tswitch m["${disc.discriminator!.property.serializedName}"] {\n`;
     for (const val of values(disc.discriminator!.all)) {
       const objSchema = <ObjectSchema>val;
-      text += `\tcase ${objSchema.discriminatorValue}:\n`;
+      let disc = objSchema.discriminatorValue;
+      // when the discriminator value is an enum, cast the const as a string
+      if (disc![0] !== '"') {
```

Comment:
This assumes the discriminator value is a string, which to be frank is likely always the case.  You could look at `disc.discriminator!.property.schema` to determine the type to cast it to although it gets a little verbose between `Choice` and `SealedChoice`.  Up to you.

Reply:
I was going to do something like what you suggested but then I saw that in the transform we're doing a check similar to the one I have above when adding the `disciminatorValue`...so I decided to leave it the same. I can switch it to the other check though, let me know if you think it's better. I also agree that I think it's basically always a string.

Reply:
Ah ok if we're doing it elsewhere then leave it as-is.